blockio-sim/src/System/FS/BlockIO/Sim.hs

```diff
@@ -21,9 +21,16 @@ fromHasFS ::
      (MonadThrow m, MonadMVar m, PrimMonad m)
   => HasFS m HandleMock
   -> m (HasBlockIO m HandleMock)
-fromHasFS = serialHasBlockIO hSetNoCache
+fromHasFS = serialHasBlockIO hSetNoCache hAdvise hAllocate
   where
+    -- TODO: It should be possible for the implementations and simulation to
+    -- throw an FsError when doing file I/O with misaligned byte arrays after
+    -- hSetNoCache. Maybe they should? However, to do that we'd have to be able
+    -- to move hSetNoCache into fs-api and fs-sim because we'd need access to
+    -- the internals.
```

Comment:
Would it really need access to the internals? I imagine one could write a wrapper `HasFS -> HasFS` that adds the alignment error checking. It'd extend the file handle type to include the tracking of the caching state. Then in the read/write it could do the alignement checks.

Reply:
Hmm I suppose you are right -- then again, it might be nicest to have everything in the same record without a wrapper